modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/SequenceSpec.scala

```diff
@@ -61,6 +62,14 @@ class SequenceSpec extends FlatSpec {
     )
   )
 
+  def runToCompletion(q: scalaz.stream.async.mutable.Queue[Event], s0: EngineState): EngineState = {
+    def isFinished(status: SequenceState): Boolean =
+      status == Idle || status == SequenceState.Completed || status === Error
+
+    q.enqueueOne(start(seqId)).flatMap( _ =>
+       processE(q).drop(1).takeThrough(a => !isFinished(a._2.get(seqId).get.status) ).runLast.eval(s0)).unsafePerformSync.get._2
```

Comment:
Yes. First I tried to use takeUntil, and I spend some time banging my head against the wall trying to understand why the resulting state was not what I expected, until I realized the state I was interested in was not included. Then I found takeThrough, and I was happy.